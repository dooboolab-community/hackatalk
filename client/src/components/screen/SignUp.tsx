import { Button, EditText } from 'dooboo-ui';
import React, { ReactElement, useState } from 'react';
import {
  SignUpMutation,
  SignUpMutationResponse,
} from '../../__generated__/SignUpMutation.graphql';
import { graphql, useMutation } from 'react-relay/hooks';
import {
  showAlertForError,
  validateEmail,
  validatePassword,
} from '../../utils/common';

import { AuthStackNavigationProps } from '../navigation/AuthStackNavigator';
import { Platform } from 'react-native';
import { ScrollView } from 'react-native-gesture-handler';
import { SignUpSendVerificationMutation } from '../../__generated__/SignUpSendVerificationMutation.graphql';
import StatusBar from '../shared/StatusBar';
import { getString } from '../../../STRINGS';
import styled from 'styled-components/native';
import { useThemeContext } from '@dooboo-ui/theme';

const Container = styled.SafeAreaView`
  flex: 1;
  background: ${({ theme }): string => theme.background};
`;

const Wrapper = styled.KeyboardAvoidingView`
  flex: 1;
`;

const ContentsWrapper = styled.View`
  margin: 44px;
`;

const ButtonWrapper = styled.View`
  width: 100%;
  margin-top: 32px;
  flex-direction: row-reverse;
`;

interface Props {
  navigation: AuthStackNavigationProps<'SignUp'>;
}

const signUp = graphql`
  mutation SignUpMutation($user: UserCreateInput!) {
    signUp(user: $user) {
      id
      email
      name
      photoURL
      verified
    }
  }
`;

const sendVerification = graphql`
  mutation SignUpSendVerificationMutation($email: String!) {
    sendVerification(email: $email)
  }
`;

function Page(props: Props): ReactElement {
  const { navigation } = props;
  const [email, setEmail] = useState<string>('');
  const [password, setPassword] = useState<string>('');
  const [confirmPassword, setConfirmPassword] = useState<string>('');
  const [name, setName] = useState<string>('');
  const [statusMessage, setStatusMessage] = useState<string>('');

  const [errorEmail, setErrorEmail] = useState<string>('');
  const [errorPassword, setErrorPassword] = useState<string>('');
  const [errorConfirmPassword, setErrorConfirmPassword] = useState<string>('');
  const [errorName, setErrorName] = useState<string>('');

  const [commitSignUp, isInFlight] = useMutation<SignUpMutation>(signUp);

  const [
    commitSendVerification,
    isVerificationInFlight,
  ] = useMutation<SignUpSendVerificationMutation>(sendVerification);

  const { theme } = useThemeContext();
  // const [signUp] = useMutation<{ signUp: AuthPayload }, MutationSignUpInput>(MUTATION_SIGN_UP);
  // const [sendVerification] =
  //   useMutation<{ sendVerification: boolean }, MutationSendVerificationInput>(MUTATION_SEND_VERIFICATION);

  const requestSignUp = async (): Promise<void> => {
    if (
      !validateEmail(email) ||
      !validatePassword(password) ||
      name.length < 2 ||
      password !== confirmPassword
    ) {
      if (!validateEmail(email)) {
        setErrorEmail(getString('EMAIL_FORMAT_NOT_VALID'));
      }

      if (!validatePassword(password)) {
        setErrorPassword(getString('PASSWORD_MIN'));
      }

      if (name.length < 2) {
        setErrorName(getString('NAME_MIN'));
      }

      if (password !== confirmPassword) {
        setErrorConfirmPassword(getString('PASSWORD_MUST_MATCH'));
      }

      return;
    }

    const mutationConfig = {
      variables: {
        user: {
          email,
          password,
          name,
          statusMessage,
        },
      },
      onCompleted: (response: SignUpMutationResponse) => {
        const sendVerificationMutationConfig = {
          variables: {
            email,
          },
        };

        commitSendVerification(sendVerificationMutationConfig);

        return navigation.replace('VerifyEmail', { email });
      },
      onError: (error: any): void => {
        showAlertForError(error);
      },
    };

    commitSignUp(mutationConfig);
  };

  const inputChangeHandlers: Record<string, (value: string) => void> = {
    emailInput: (email: string): void => {
      setEmail(email);

      if (!validateEmail(email)) {
        setErrorEmail(getString('EMAIL_FORMAT_NOT_VALID'));
      } else {
        setErrorEmail('');
      }
    },
    passwordInput: (password: string): void => {
      setPassword(password);

      if (!validatePassword(password)) {
        setErrorPassword(getString('PASSWORD_MIN'));
      } else if (confirmPassword && password !== confirmPassword) {
        setErrorPassword('');
        setErrorConfirmPassword(getString('PASSWORD_MUST_MATCH'));
      } else {
        setErrorPassword('');
      }
    },
    confirmPasswordInput: (confirmPassword: string): void => {
      setConfirmPassword(confirmPassword);

      if (password !== confirmPassword) {
        setErrorConfirmPassword(getString('PASSWORD_MUST_MATCH'));
      } else {
        setErrorConfirmPassword('');
      }
    },
    nameInput: (name: string): void => {
      setName(name);

      if (name.length < 2) {
        setErrorName(getString('NAME_MIN'));
      } else {
        setErrorName('');
      }
    },
  };

  return (
    <Container>
      <StatusBar />
      <Wrapper
        behavior={Platform.select({
          ios: 'padding',
          android: 'height',
        })}
        keyboardVerticalOffset={100}
      >
        <ScrollView style={{ alignSelf: 'stretch' }}>
          <ContentsWrapper>
            <EditText
              testID="input-email"
              errorTestID="error-email"
              textStyle={{
                color: theme.fontColor,
              }}
              borderColor={theme.font}
              focusColor={theme.focused}
              placeholderTextColor={theme.placeholder}
              label={getString('EMAIL')}
              placeholder="hello@example.com"
              value={email}
              onChangeText={inputChangeHandlers.emailInput}
              errorText={errorEmail}
              onSubmitEditing={requestSignUp}
            />
            <EditText
              testID="input-password"
              errorTestID="error-password"
              textStyle={{
                color: theme.fontColor,
              }}
              borderColor={theme.font}
              focusColor={theme.focused}
              placeholderTextColor={theme.placeholder}
              placeholder="********"
              label={getString('PASSWORD')}
              value={password}
              onChangeText={inputChangeHandlers.passwordInput}
              style={{ marginTop: 32 }}
              errorText={errorPassword}
              onSubmitEditing={requestSignUp}
              secureTextEntry={true}
            />
            <EditText
              testID="input-confirm-password"
              errorTestID="error-confirm-password"
              textStyle={{
                color: theme.fontColor,
              }}
              placeholder="********"
              label={getString('CONFIRM_PASSWORD')}
              value={confirmPassword}
              onChangeText={inputChangeHandlers.confirmPasswordInput}
              style={{ marginTop: 32 }}
              borderColor={theme.font}
              focusColor={theme.focused}
              placeholderTextColor={theme.placeholder}
              errorText={errorConfirmPassword}
              onSubmitEditing={requestSignUp}
              secureTextEntry={true}
            />
            <EditText
              testID="input-name"
              errorTestID="error-name"
              textStyle={{
                color: theme.fontColor,
              }}
              label={getString('NAME')}
              placeholder={getString('NAME_HINT')}
              borderColor={theme.font}
              focusColor={theme.focused}
              placeholderTextColor={theme.placeholder}
              value={name}
              onChangeText={inputChangeHandlers.nameInput}
              style={{ marginTop: 32 }}
              errorText={errorName}
              onSubmitEditing={requestSignUp}
            />
            <EditText
              testID="input-status"
              errorTestID="error-status"
              textStyle={{
                color: theme.fontColor,
              }}
              borderColor={theme.font}
              focusColor={theme.focused}
              placeholderTextColor={theme.placeholder}
              label={getString('STATUS')}
              placeholder={getString('STATUS_MSG_HINT')}
              value={statusMessage}
              onChangeText={(text: string): void => {
                setStatusMessage(text);
              }}
              style={{ marginTop: 32 }}
              onSubmitEditing={requestSignUp}
            />
            <ButtonWrapper>
              <Button
                testID="btn-sign-up"
                loading={isInFlight}
                onPress={requestSignUp}
                style={{
                  root: {
                    height: 52,
                    width: '50%',
                    backgroundColor: theme.btnPrimary,
                  },
                  text: {
                    color: theme.btnPrimaryFont,
                    fontSize: 16,
                  },
                }}
                text={getString('SIGN_UP')}
              />
            </ButtonWrapper>
          </ContentsWrapper>
        </ScrollView>
      </Wrapper>
    </Container>
  );
}

export default Page;
