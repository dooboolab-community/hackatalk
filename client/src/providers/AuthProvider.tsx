import React, {useReducer} from 'react';

import {User} from '../types/graphql';
import createCtx from '../utils/createCtx';

export interface State {
  user: User | undefined;
}

interface Context {
  state: State;
  setUser(user: User | undefined): void;
}

const [useCtx, Provider] = createCtx<Context>();

// eslint-disable-next-line no-shadow
export enum ActionType {
  SetUser = 'set-user',
}

type SetUserAction = {
  type: ActionType.SetUser;
  payload: {user: User};
};
type Action = SetUserAction;

interface Props {
  children?: React.ReactElement;
  initialAuthUser?: User;
}

type Reducer = (state: State, action: Action) => State;

const setUser = (dispatch: React.Dispatch<SetUserAction>) => (
  authUser: User,
): void => {
  dispatch({
    type: ActionType.SetUser,
    payload: {user: authUser},
  });
};

const initialState: State = {
  user: undefined,
};

// eslint-disable-next-line default-param-last
const reducer: Reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SetUser:
      return {...action.payload};
    default:
      return state;
  }
};

function AuthProvider(props: Props): React.ReactElement {
  const initialAuthState = {
    user: props.initialAuthUser,
  };

  const [state, dispatch] = useReducer<Reducer>(reducer, initialAuthState);

  const actions = {
    setUser: setUser(dispatch),
  };

  return <Provider value={{state, ...actions}}>{props.children}</Provider>;
}

const AuthContext = {
  useAuthContext: useCtx,
  AuthProvider,
};

export {useCtx as useAuthContext, AuthProvider};
export default AuthContext;
