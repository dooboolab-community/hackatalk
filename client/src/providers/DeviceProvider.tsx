import * as Device from 'expo-device';

import React, {FC, useEffect, useState} from 'react';

import createCtx from '../utils/createCtx';

interface Context {
  deviceType: Device.DeviceType;
  setDeviceType: React.Dispatch<React.SetStateAction<Device.DeviceType>>;
}

const [useCtx, Provider] = createCtx<Context>();

const DeviceProvider: FC = ({children}) => {
  const [deviceType, setDeviceType] = useState<Device.DeviceType>(
    Device.DeviceType.PHONE,
  );

  useEffect(() => {
    let isMounted = true;

    Device.getDeviceTypeAsync().then((result) => {
      // Only change state if this component is still mounted.
      if (isMounted) setDeviceType(result);
    });

    return () => {
      isMounted = false;
    };
  }, []);

  return (
    <Provider
      value={{
        deviceType,
        setDeviceType,
      }}>
      {children}
    </Provider>
  );
};

const DeviceContext = {
  useDeviceContext: useCtx,
  DeviceProvider,
};

export {useCtx as useDeviceContext, DeviceProvider};
export default DeviceContext;
