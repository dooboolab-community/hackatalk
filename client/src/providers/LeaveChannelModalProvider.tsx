import React, {createContext, useContext, useState} from 'react';
import type {ReactElement, ReactNode} from 'react';

type LeaveChannelModalData = {
  testID?: string;
  channelId: string;
  onClickYes?: () => void;
};

export type LeaveChannelModalState =
  | {isVisible: false}
  | ({isVisible: true} & LeaveChannelModalData);

export type LeaveChannelModalContext = {
  showModal: (next: LeaveChannelModalData) => void;
  hideModal: () => void;
  modalState: LeaveChannelModalState;
};

const LeaveChannelModalContext = createContext<LeaveChannelModalContext>({
  modalState: {isVisible: false},
  showModal: () => {},
  hideModal: () => {},
});

type Props = {
  children: ReactNode;
};

export const LeaveChannelModalProvider = ({children}: Props): ReactElement => {
  const [showModalState, setModalState] = useState<LeaveChannelModalState>({
    isVisible: false,
  });

  const showModal = (next: LeaveChannelModalData): void =>
    setModalState({
      isVisible: true,
      ...next,
    });

  const hideModal = (): void =>
    setModalState({
      isVisible: false,
    });

  return (
    <LeaveChannelModalContext.Provider
      value={{
        showModal,
        hideModal,
        modalState: showModalState,
      }}
    >
      {children}
    </LeaveChannelModalContext.Provider>
  );
};

export const useLeaveChannelContext = (): LeaveChannelModalContext =>
  useContext(LeaveChannelModalContext);
