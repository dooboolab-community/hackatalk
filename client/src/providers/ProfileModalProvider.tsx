import React, {FC, createContext, useContext, useState} from 'react';

import {ProfileModal_user$key} from '../__generated__/ProfileModal_user.graphql';

export type ModalState =
  | {
      isVisible: false;
    }
  | {
      /** Modal visiblity */
      isVisible: true;
      /** Which user the profile modal describes */
      user: ProfileModal_user$key;
      /** Is the profile user a friend of the current user? */
      isFriend?: boolean;
      /** Callback function for delete button */
      onDeleteFriend?: () => void;
      /** Callback function for add friend button */
      onAddFriend?: () => void;
      hideButtons?: boolean;
    };

export type ProfileModalContext = {
  /** Allow context consumers to make the modal visible with a new state. */
  showModal: (next: Omit<ModalState & {isVisible: true}, 'isVisible'>) => void;
  /** Allow context consumers to hide the modal. */
  hideModal: () => void;
  modalState: ModalState;
};

const ProfileModalContext = createContext<ProfileModalContext>({
  modalState: {isVisible: false},
  showModal: () => {},
  hideModal: () => {},
});

export const ProfileModalProvider: FC = (props) => {
  const [state, setState] = useState<ModalState>({
    isVisible: false,
  });

  const showModal = (
    next: Omit<ModalState & {isVisible: true}, 'isVisible'>,
  ): void =>
    setState({
      isVisible: true,
      ...next,
    });

  const hideModal = (): void =>
    setState({
      isVisible: false,
    });

  return (
    <ProfileModalContext.Provider
      value={{
        showModal,
        hideModal,
        modalState: state,
      }}>
      {props.children}
    </ProfileModalContext.Provider>
  );
};

export const useProfileContext = (): ProfileModalContext =>
  useContext(ProfileModalContext);
