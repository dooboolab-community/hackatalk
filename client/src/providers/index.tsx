export * from './AuthProvider';
export * from './DeviceProvider';
export * from './ProfileModalProvider';
