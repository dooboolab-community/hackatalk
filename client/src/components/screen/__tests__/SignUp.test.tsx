import 'react-native';

import {
  cleanup,
  fireEvent,
  render,
  waitFor,
} from '@testing-library/react-native';
import {createTestElement, createTestProps} from '../../../../test/testUtils';

import {MockPayloadGenerator} from 'relay-test-utils';
import React from 'react';
import SignUp from '../SignUp';
import {environment} from '../../../providers';

const component = createTestElement(<SignUp {...createTestProps()} />);

// const mockSignUpMutation = [
//   {
//     request: {
//       query: MUTATION_SIGN_UP,
//       variables: {
//         user: {
//           email: 'test@email.com',
//           name: 'name',
//           password: 'testpass12!',
//           statusMessage: 'status',
//         },
//       },
//     },
//     result: {
//       data: {
//         signUp: {
//           token: 'access token',
//           user: {
//             id: 'userId',
//             email: 'test@email.com',
//             nickname: 'nickname',
//             statusMessage: 'status',
//             verified: false,
//           },
//         },
//       },
//     },
//   },
// ];

describe('[SignUp] rendering test', () => {
  it('renders as expected', () => {
    const json = render(component).toJSON();

    expect(json).toBeTruthy();
    expect(json).toMatchSnapshot();
  });
});

describe('[SignUp] interaction', () => {
  it('should invoke changeText event handler when email changed', async () => {
    const {getByTestId} = render(component);
    const textInput = getByTestId('input-email');

    await waitFor(() => expect(textInput).toBeTruthy());

    fireEvent.changeText(textInput, 'email@email.com');

    expect(textInput.props.value).toEqual('email@email.com');
  });

  it('should invoke changeText event handler when password changed', async () => {
    const {getByTestId} = render(component);

    const textInput = getByTestId('input-password');

    await waitFor(() => expect(textInput).toBeTruthy());

    fireEvent.changeText(textInput, 'pw test');

    expect(textInput.props.value).toEqual('pw test');
  });

  it('should confirm password', async () => {
    const {getByTestId} = render(component);

    const textInput = getByTestId('input-confirm-password');

    await waitFor(() => expect(textInput).toBeTruthy());

    fireEvent.changeText(textInput, 'pw test');

    expect(textInput.props.value).toEqual('pw test');
  });

  it('should invoke changeText event when name changed', async () => {
    const {getByTestId} = render(component);
    const textInput = getByTestId('input-name');

    await waitFor(() => expect(textInput).toBeTruthy());

    fireEvent.changeText(textInput, 'dooboo');

    expect(textInput.props.value).toEqual('dooboo');
  });

  it('should invoke changeText event when status changed', async () => {
    const {getByTestId} = render(component);
    const textInput = getByTestId('input-status');

    await waitFor(() => expect(textInput).toBeTruthy());

    fireEvent.changeText(textInput, 'I am good');

    expect(textInput.props.value).toEqual('I am good');
  });

  // describe('onSignUp', () => {
  //   describe('check validation', () => {
  //     it('should show error text when the email is not validated', async () => {
  //       const {getByTestId} = render(component);
  //       const emailInput = getByTestId('input-email');

  //       await waitFor(() => expect(emailInput).toBeTruthy());

  //       fireEvent.changeText(emailInput, 'email@email');

  //       const passwordInput = getByTestId('input-password');

  //       await waitFor(() => expect(emailInput).toBeTruthy());

  //       fireEvent.changeText(passwordInput, 'Abc123##');

  //       const confirmPasswordInput = getByTestId('input-confirm-password');

  //       await waitFor(() => expect(confirmPasswordInput).toBeTruthy());

  //       fireEvent.changeText(confirmPasswordInput, 'Abc123##');

  //       const nameInput = getByTestId('input-name');

  //       await waitFor(() => expect(nameInput).toBeTruthy());

  //       fireEvent.changeText(nameInput, 'dooboo');

  //       const btnSignUp = getByTestId('btn-sign-up');

  //       await waitFor(() => expect(btnSignUp).toBeTruthy());

  //       fireEvent.press(btnSignUp);

  //       const errorText = getByTestId('error-email');

  //       await waitFor(() => expect(errorText).toBeTruthy());
  //     });

  //     it('should show error text when the password is not validated', async () => {
  //       const {getByTestId} = render(component);
  //       const emailInput = getByTestId('input-email');

  //       await waitFor(() => expect(emailInput).toBeTruthy());

  //       fireEvent.changeText(emailInput, 'email@email.com');

  //       const passwordInput = getByTestId('input-password');

  //       await waitFor(() => expect(emailInput).toBeTruthy());

  //       fireEvent.changeText(passwordInput, 'Abc');

  //       const confirmPasswordInput = getByTestId('input-confirm-password');

  //       await waitFor(() => expect(confirmPasswordInput).toBeTruthy());

  //       fireEvent.changeText(confirmPasswordInput, 'Abc');

  //       const nameInput = getByTestId('input-name');

  //       await waitFor(() => expect(nameInput).toBeTruthy());

  //       fireEvent.changeText(nameInput, 'dooboo');

  //       const btnSignUp = getByTestId('btn-sign-up');

  //       await waitFor(() => expect(btnSignUp).toBeTruthy());

  //       fireEvent.press(btnSignUp);

  //       const errorText = getByTestId('error-password');

  //       await waitFor(() => expect(errorText).toBeTruthy());
  //     });
  //   });

  //   it('should show error text when the name is not validated', async () => {
  //     const {getByTestId} = render(component);
  //     const emailInput = getByTestId('input-email');

  //     await waitFor(() => expect(emailInput).toBeTruthy());

  //     fireEvent.changeText(emailInput, 'email@email.com');

  //     const passwordInput = getByTestId('input-password');

  //     await waitFor(() => expect(emailInput).toBeTruthy());

  //     fireEvent.changeText(passwordInput, 'Abc123##');

  //     const confirmPasswordInput = getByTestId('input-confirm-password');

  //     await waitFor(() => expect(confirmPasswordInput).toBeTruthy());

  //     fireEvent.changeText(confirmPasswordInput, 'Abc123##');

  //     const nameInput = getByTestId('input-name');

  //     await waitFor(() => expect(nameInput).toBeTruthy());

  //     fireEvent.changeText(nameInput, '');

  //     const btnSignUp = getByTestId('btn-sign-up');

  //     await waitFor(() => expect(btnSignUp).toBeTruthy());

  //     fireEvent.press(btnSignUp);

  //     const errorText = getByTestId('error-name');

  //     await waitFor(() => expect(errorText).toBeTruthy());
  //   });

  //   it('should show error text when the password is not confirmed', async () => {
  //     const {getByTestId} = render(component);
  //     const emailInput = getByTestId('input-email');

  //     await waitFor(() => expect(emailInput).toBeTruthy());

  //     fireEvent.changeText(emailInput, 'email@email.com');

  //     const passwordInput = getByTestId('input-password');

  //     await waitFor(() => expect(emailInput).toBeTruthy());

  //     fireEvent.changeText(passwordInput, 'Abc123##');

  //     const confirmPasswordInput = getByTestId('input-confirm-password');

  //     await waitFor(() => expect(confirmPasswordInput).toBeTruthy());

  //     fireEvent.changeText(confirmPasswordInput, 'Adjdj123??');

  //     const nameInput = getByTestId('input-name');

  //     await waitFor(() => expect(nameInput).toBeTruthy());

  //     fireEvent.changeText(nameInput, 'dooboo');

  //     const btnSignUp = getByTestId('btn-sign-up');

  //     await waitFor(() => expect(btnSignUp).toBeTruthy());

  //     fireEvent.press(btnSignUp);

  //     const errorText = getByTestId('error-confirm-password');

  //     await waitFor(() => expect(errorText).toBeTruthy());
  //   });
  // });

  it('should call signUp when button has clicked and navigate to MainStack', async () => {
    const {getByTestId} = render(component);
    const emailInput = getByTestId('input-email');

    await waitFor(() => expect(emailInput).toBeTruthy());

    fireEvent.changeText(emailInput, 'test@email.com');

    const passwordInput = getByTestId('input-password');

    await waitFor(() => expect(passwordInput).toBeTruthy());

    fireEvent.changeText(passwordInput, 'testpass12!');

    const confirmPasswordInput = getByTestId('input-confirm-password');

    await waitFor(() => expect(confirmPasswordInput).toBeTruthy());

    fireEvent.changeText(confirmPasswordInput, 'testpass12!');

    const nameInput = getByTestId('input-name');

    await waitFor(() => expect(nameInput).toBeTruthy());

    fireEvent.changeText(nameInput, 'name');

    const statusInput = getByTestId('input-status');

    await waitFor(() => expect(nameInput).toBeTruthy());

    fireEvent.changeText(statusInput, 'status');

    const btnSignUp = getByTestId('btn-sign-up');

    await waitFor(() => expect(btnSignUp).toBeTruthy());

    fireEvent.press(btnSignUp);

    const operation = environment.mock.getMostRecentOperation();

    environment.mock.resolve(
      operation,
      MockPayloadGenerator.generate(operation),
    );
    // expect(props.navigation.resetRoot).toHaveBeenCalledTimes(1);
  });

  it('should call signUp when the button has clicked and check whether it catches error', async () => {
    const anotherComponent = createTestElement(
      <SignUp
        {...createTestProps({
          navigation: null,
        })}
      />,
    );

    const {getByTestId} = render(anotherComponent);

    const emailInput = getByTestId('input-email');

    await waitFor(() => expect(emailInput).toBeTruthy());

    fireEvent.changeText(emailInput, 'email@email.com');

    const passwordInput = getByTestId('input-password');

    await waitFor(() => expect(passwordInput).toBeTruthy());

    fireEvent.changeText(passwordInput, 'Abc123##');

    const confirmPasswordInput = getByTestId('input-confirm-password');

    await waitFor(() => expect(confirmPasswordInput).toBeTruthy());

    fireEvent.changeText(confirmPasswordInput, 'Abc123##');

    const nameInput = getByTestId('input-name');

    await waitFor(() => expect(nameInput).toBeTruthy());

    fireEvent.changeText(nameInput, 'dooboo');

    const btnSignUp = getByTestId('btn-sign-up');

    await waitFor(() => expect(btnSignUp).toBeTruthy());

    fireEvent.press(btnSignUp);
  });

  afterAll((done) => {
    cleanup();
    done();
  });
});
