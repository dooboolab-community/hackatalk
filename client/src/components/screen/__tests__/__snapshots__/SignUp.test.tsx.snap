// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SignUp] rendering test renders as expected 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ],
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
        }
      }
    >
      <View>
        <View
          style={
            Array [
              Object {
                "marginBottom": 44,
                "marginLeft": 44,
                "marginRight": 44,
                "marginTop": 44,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#b9b9c4",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "marginBottom": 5,
                  },
                  undefined,
                  false,
                ]
              }
            >
              Email
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              placeholder="hello@example.com"
              placeholderTextColor="#676c7a"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "fontSize": 15,
                    "fontWeight": "500",
                    "minHeight": 44,
                    "paddingBottom": 15,
                    "paddingTop": 15,
                  },
                  Object {
                    "color": "#151A25",
                  },
                ]
              }
              testID="input-email"
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "width": "100%",
                  },
                  undefined,
                  Object {
                    "borderBottomWidth": 0.6,
                    "borderColor": "#2E2E2E",
                  },
                  false,
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                Object {
                  "marginTop": 32,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#b9b9c4",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "marginBottom": 5,
                  },
                  undefined,
                  false,
                ]
              }
            >
              Password
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              placeholder="********"
              placeholderTextColor="#676c7a"
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Array [
                  Object {
                    "fontSize": 15,
                    "fontWeight": "500",
                    "minHeight": 44,
                    "paddingBottom": 15,
                    "paddingTop": 15,
                  },
                  Object {
                    "color": "#151A25",
                  },
                ]
              }
              testID="input-password"
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "width": "100%",
                  },
                  undefined,
                  Object {
                    "borderBottomWidth": 0.6,
                    "borderColor": "#2E2E2E",
                  },
                  false,
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                Object {
                  "marginTop": 32,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#b9b9c4",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "marginBottom": 5,
                  },
                  undefined,
                  false,
                ]
              }
            >
              Confirm password
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              placeholder="********"
              placeholderTextColor="#676c7a"
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Array [
                  Object {
                    "fontSize": 15,
                    "fontWeight": "500",
                    "minHeight": 44,
                    "paddingBottom": 15,
                    "paddingTop": 15,
                  },
                  Object {
                    "color": "#151A25",
                  },
                ]
              }
              testID="input-confirm-password"
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "width": "100%",
                  },
                  undefined,
                  Object {
                    "borderBottomWidth": 0.6,
                    "borderColor": "#2E2E2E",
                  },
                  false,
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                Object {
                  "marginTop": 32,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#b9b9c4",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "marginBottom": 5,
                  },
                  undefined,
                  false,
                ]
              }
            >
              Name
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Please write your name"
              placeholderTextColor="#676c7a"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "fontSize": 15,
                    "fontWeight": "500",
                    "minHeight": 44,
                    "paddingBottom": 15,
                    "paddingTop": 15,
                  },
                  Object {
                    "color": "#151A25",
                  },
                ]
              }
              testID="input-name"
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "width": "100%",
                  },
                  undefined,
                  Object {
                    "borderBottomWidth": 0.6,
                    "borderColor": "#2E2E2E",
                  },
                  false,
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                Object {
                  "marginTop": 32,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#b9b9c4",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "marginBottom": 5,
                  },
                  undefined,
                  false,
                ]
              }
            >
              Status
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Please write your status message"
              placeholderTextColor="#676c7a"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "fontSize": 15,
                    "fontWeight": "500",
                    "minHeight": 44,
                    "paddingBottom": 15,
                    "paddingTop": 15,
                  },
                  Object {
                    "color": "#151A25",
                  },
                ]
              }
              testID="input-status"
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "width": "100%",
                  },
                  undefined,
                  Object {
                    "borderBottomWidth": 0.6,
                    "borderColor": "#2E2E2E",
                  },
                  false,
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row-reverse",
                  "marginTop": 32,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="btn-sign-up"
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
